src/NLog.Web.AspNetCore/NLogBufferingMiddleware.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using NLog.Web.Targets.Wrappers;
+
+namespace NLog.Web
+{
+    /// <summary>
+    /// This class is to intercept the HTTP pipeline and to allow additional logging of the following
+    ///
+    /// POST request body
+    ///
+    /// Usage: app.UseMiddleware&lt;NLogBufferingMiddleware&gt;(); where app is an IApplicationBuilder
+    /// </summary>
+    public class NLogBufferingMiddleware
```

Comment:
Think the class-name is a little vague.

Reply:
Renamed to NLogBufferingTargetWrapperMiddleware